packages/salesforcedx-utils-vscode/src/cli/commandExecutor.ts

```diff
@@ -58,6 +74,64 @@ export interface CommandExecution {
   readonly stderrSubject: Observable<Buffer | string>;
 }
 
+export class CompositeCliCommandExecution implements CommandExecution {
+  public readonly exitSubject: Subject<number | undefined>;
+  public readonly errorSubject: Subject<Error | undefined>;
+  public readonly command: Command;
+  public readonly cancellationToken?: CancellationToken;
+  public readonly processExitSubject: Observable<number | undefined>;
+  public readonly processErrorSubject: Observable<Error | undefined>;
+  public readonly stdoutSubject: Observable<Buffer | string>;
+  public readonly stderrSubject: Observable<Buffer | string>;
+
+  constructor(command: Command, cancellationToken?: CancellationToken) {
+    this.exitSubject = new Subject();
+    this.errorSubject = new Subject();
+    this.command = command;
+    this.cancellationToken = cancellationToken;
+    this.processExitSubject = this.exitSubject.asObservable();
+    this.processErrorSubject = this.errorSubject.asObservable();
+    this.stdoutSubject = EmptyObservable.create();
+    this.stderrSubject = EmptyObservable.create();
+
+    let timerSubscriber: Subscription | null;
+    if (cancellationToken) {
+      const timer = Observable.interval(1000);
+      timerSubscriber = timer.subscribe(async next => {
+        if (cancellationToken.isCancellationRequested) {
+          try {
+            this.exitSubject.next();
+          } catch (e) {
+            console.log(e);
+          }
+        }
+      });
+    }
+    this.processErrorSubject.subscribe(next => {
+      if (timerSubscriber) {
+        timerSubscriber.unsubscribe();
+      }
+    });
+
+    this.processExitSubject.subscribe(next => {
+      if (timerSubscriber) {
+        timerSubscriber.unsubscribe();
+      }
+    });
+  }
+
+  public successfulExit() {
+    this.exitSubject.next(0);
+  }
+
+  public failureExit(e?: any) {
+    this.exitSubject.next(1);
+    if (e) {
+      console.log(e);
```

Comment:
I don't quite understand why you are just logging to console here. Does this mean that the user doesn't get to use the error displayed in the UI if an error were to occur?